### PR TITLE
[BEAM-1778] First clean up pass of dataflow references/URLs in Java SDK

### PR DESCRIPTION
**Pull request by melap**
R: @aaltay

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java [hunk 1]
```diff
@@ -122,7 +120,7 @@
  *
  * <h3>Permissions</h3>
  * Permission requirements depend on the {@link PipelineRunner} that is used to execute the
- * Dataflow job. Please refer to the documentation of corresponding {@link PipelineRunner}s for
+ * job. Please refer to the documentation of corresponding {@link PipelineRunner}s for
```
Comment (aaltay) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroIO.java]: `job` -> `pipeline` ?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/CloudResourceManagerOptions.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 import org.apache.beam.sdk.util.GcpProjectUtil;
 
 /**
- * Properties needed when using CloudResourceManager with the Beam SDK.
+ * Properties needed when using Google CloudResourceManager with the Beam SDK.
  */
 @Description("Options that are used to configure CloudResourceManager. See "
```
Comment (aaltay) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/CloudResourceManagerOptions.java]: `CloudResourceManager` -> `Google CloudResourceManager`?

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java [hunk 3]
```diff
@@ -112,7 +112,7 @@
   void setPathValidator(PathValidator validator);
 
   /**
-   * Returns the default {@link ExecutorService} to use within the Dataflow SDK. The
+   * Returns the default {@link ExecutorService} to use within the Beam SDK. The
```
Comment (aaltay) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/GcsOptions.java]: `Apache Beam SDK`

File: sdks/java/java8tests/src/test/java/org/apache/beam/sdk/transforms/MapElementsJava8Test.java [hunk 4]
```diff
@@ -46,7 +46,7 @@ public void testMapLambda() throws Exception {
     PCollection<Integer> output = pipeline
         .apply(Create.of(1, 2, 3))
         .apply(MapElements
-            // Note that the type annotation is required (for Java, not for Dataflow)
```
Comment (aaltay) [inline sdks/java/java8tests/src/test/java/org/apache/beam/sdk/transforms/MapElementsJava8Test.java]: ? `for java, not for Apache Beam`, or alternatively drop the whole parenthesis.

Comment (melap) [inline sdks/java/java8tests/src/test/java/org/apache/beam/sdk/transforms/MapElementsJava8Test.java]: dropped the paren

**Comment by aaltay** [comment]
LGTM. Thank you.

**Comment by melap** [comment]
Thanks! I've incorporated your feedback.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/options/GoogleApiDebugOptions.java [hunk 5]
```diff
@@ -40,7 +40,7 @@
    * to Google, specifically to the Google Cloud Dataflow team.
    */
   @Description("This option enables tracing of API calls to Google services used within the "
-      + "Dataflow SDK. Values are expected in JSON format {\"ApiName\":\"TraceDestination\",...} "
+      + "Apache Beam SDK. Values are expected in JSON format {\"ApiName\":\"TraceDestination\",...} "
```
Comment (aaltay) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/GoogleApiDebugOptions.java]: There is a lint error here:
GoogleApiDebugOptions.java:43: Line is longer than 100 characters (found 101). [LineLength]

perhaps move the empty space at the end of the line to the next line.

Comment (melap) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/options/GoogleApiDebugOptions.java]: moved Apache up a line so spaces stay confined to the end of the lines, still squeaks by at 100 chars

**Pull request closed without merging**